tests/helpers/test_entity_component.py

```diff
@@ -556,6 +556,26 @@ def appender(**kwargs):
     assert len(calls) == 2
 
 
+async def test_register_entity_service_non_entity_service_schema(
+    hass: HomeAssistant,
+) -> None:
+    """Test attempting to register a service with an incomplete schema."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    with pytest.raises(
+        HomeAssistantError,
+        match=(
+            "The schema does not include all required keys: entity_id, device_id, area_id, "
+            "floor_id, label_id"
+        ),
+    ):
+        component.async_register_entity_service(
+            "hello",
+            vol.Schema({"some": str}),
```

Comment:
May be parameterize some more tests including schema's with `vol.All` and `vol.Any` ?